Python/Product/Debugger/DebugAdapterProcess.cs

```diff
@@ -64,5 +135,122 @@ public sealed class DebugAdapterProcess : ITargetHostProcess {
         }
 
         public void Terminate() => _process.Kill();
+
+        private void InitializeListenerSocket() {
+            if (_listenerPort < 0) {
+                var socketSource = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.IP);
+                socketSource.Bind(new IPEndPoint(IPAddress.Loopback, 0));
+                socketSource.Listen(0);
+                _listenerPort = ((IPEndPoint)socketSource.LocalEndPoint).Port;
+                Debug.WriteLine("Listening for debug connections on port {0}", _listenerPort);
+                socketSource.BeginAccept(AcceptConnection, socketSource);
+            }
+        }
+
+        private void AcceptConnection(IAsyncResult iar) {
+            var socketSource = ((Socket)iar.AsyncState);
+            try {
+                _socket = socketSource.EndAccept(iar);
+            } catch (SocketException ex) {
+                Debug.WriteLine("Debug Adapter Host socket failed");
+                Debug.WriteLine(ex);
+            } catch (ObjectDisposedException) {
+                Debug.WriteLine("Debug Adapter Host socket closed");
+            }
+
+            if (_socket != null) {
+                _stream = new NetworkStream(_socket, ownsSocket: true);
+                _connectedEvent.Set();
+            }
+        }
+
+        
+
+        /***************************************************************************************************************
+        * The code below should go into refactored PythonDebugOptions class. See AD7Engine ParseOptions for more info
+        ****************************************************************************************************************/
+
+
+        private static string[] SplitOptions(string options) {
+            List<string> res = new List<string>();
+            int lastStart = 0;
+            for (int i = 0; i < options.Length; i++) {
+                if (options[i] == ';') {
+                    if (i < options.Length - 1 && options[i + 1] != ';') {
+                        // valid option boundary
+                        res.Add(options.Substring(lastStart, i - lastStart));
+                        lastStart = i + 1;
+                    } else {
+                        i++;
```

Comment:
Why do you need this?

Reply:
This code is taken as is from `AD7Engine`. This particular `if` just skips over any redundant `;`, I think.